grails-app/services/org/pih/warehouse/importer/ProductSupplierImportDataService.groovy

```diff
@@ -47,7 +48,18 @@ class ProductSupplierImportDataService implements ImportDataService {
                 command.errors.reject("Row ${index + 1}: Supplier Name is required")
             }
 
-            if (params.supplierName && !Organization.findByName(params.supplierName)) {
+            Organization supplier = params.supplierName ? Organization.findByName(params.supplierName) : null
+
+            /**
+             *  Prevent from assigning an inactive supplier to a source that is about to become active or while creating a new source
+                "to become active" means that an existing source is active and we don't change it, or a source is inactive and we are activating it during the import
+                Allow assigning an inactive supplier to an inactive source (or to a source that is about to become inactive)
+             */
+            if (supplier && !supplier.active && (params.active || !productSupplierExists)) {
+                command.errors.reject("Row ${index + 1}: Supplier '${supplier.name}' is no longer active. Choose an active supplier")
```

Comment:
Does merging the two cases (ProductSupplier does not exist and Organization is inactive) into the same error message give us enough context on the frontend to know what to do? Seems like for the first case we need to change params.id, and for the second we need to change params.supplierName or params.active

Reply:
the existence of the product supplier only indicates whether we are editing a source or creating a brand new one. Since the requirements are different for editing and creating a brand new one (explained in the comment above the if), I needed this indication whether the source exists or not.